modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/http4s/WebServerLauncher.scala

```diff
@@ -1,38 +1,111 @@
 package edu.gemini.seqexec.web.server.http4s
 
+import java.io.File
 import java.util.logging.Logger
 
+import edu.gemini.seqexec.server.ExecutorImpl
 import edu.gemini.seqexec.web.server.common.LogInitialization
-import org.http4s.server.Server
+import edu.gemini.seqexec.web.server.security.{AuthenticationConfig, AuthenticationService, LDAPConfig}
+import edu.gemini.spModel.core.Peer
+import knobs._
+import org.http4s.server.{Server, ServerApp}
 import org.http4s.server.blaze.BlazeBuilder
 
 import scalaz._
 import Scalaz._
+import scalaz.concurrent.Task
 import scalaz.effect.IO
 
-object WebServerLauncher extends App with LogInitialization {
+object WebServerLauncher extends ServerApp with LogInitialization {
   // Initialize the log and exit if it fails
   configLog.run.onException(IO(sys.exit(1))).unsafePerformIO()
 
+  // Initialize logger after the configuration
   val logger = Logger.getLogger(getClass.getName)
 
-  // TODO improve configuration style
-  val devMode = !args.contains("prod")
-
-  def launch(port: Int):Option[Server] = {
-    logger.info(s"Starting web server on port $port")
-    try {
-      Some(BlazeBuilder.bindHttp(port, "0.0.0.0")
-        .withWebSockets(true)
-        .mountService(StaticRoutes.service(devMode), "/")
-        .mountService(SeqexecCommandRoutes.service, "/api/seqexec/commands")
-        .mountService(SeqexecUIApiRoutes.service, "/api")
-        .run)
-    } catch {
-      case e: Throwable =>
-        e.printStackTrace()
-        None
+  /**
+    * Configuration for the web server
+    */
+  case class WebServerConfiguration(host: String, port: Int, devMode: Boolean)
+
+  /**
+    * Configuration for the seqexec engine
+    */
+  case class SeqexecConfiguration(odbHost: String)
+
+  // Attempt to get the file or throw an exception if not possible
+  val configurationFile: File = new File(new File(baseDir, "conf"), "app.conf")
+
+  // Read the config, first attempt the file or default to the classpath file
+  val config: Task[Config] = knobs.loadImmutable(
+    Required(FileResource(configurationFile) or ClassPathResource("app.conf")) :: Nil)
```

Comment:
Knobs has support for checking config file changes and support reloading, but I didn't want to go there. I think it is a valid point, I can imagine cases where you want to reload the config. On the other hand it worries me that somebody will edit the configuration and do a restart it right away by mistake

In general we have tended to have configs set in stone, even inside the jar files to avoid accidental editing but I can see how it could be useful in some cases.